### PR TITLE
syncer: move UpstreamConn into dbconn package

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

too many files under `syncer` package

### What is changed and how it works?

mainly syncer/db.go → syncer/dbconn/upstream_db.go

### Check List 

Tests 

 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: dm/config/subtask.go [hunk 1]
```diff
@@ -122,6 +124,29 @@ func (db *DBConfig) Adjust() {
 	AdjustTargetDBTimeZone(db)
 }
 
+// GetDBConfigFromEnv is a helper function to read config from environment. It's commonly used in unit tests.
+func GetDBConfigFromEnv() DBConfig {
```
Comment (lance6716) [inline dm/config/subtask.go:128]: this function is duplicated in many place originall

**Comment by lichunzhu** [comment]
Is this PR ready for review?

**Comment by lance6716** [comment]
> Is this PR ready for review?

yes

File: syncer/syncer_test.go [hunk 2]
```diff
@@ -1839,3 +1844,176 @@ func (s *testSyncerSuite) TestDownstreamTableHasAutoRandom(c *C) {
...
+		Cfg: &config.SubTaskConfig{
+			Name: "test",
+		},
+	}
+
+	sqls := []string{"alter table t1 add column a int", "alter table t1 add column b int"}
+
+	// will ignore the first error, and continue execute the second sql
+	mock.ExpectBegin()
+	mock.ExpectExec("alter table t1 add column a int").WillReturnError(newMysqlErr(uint16(infoschema.ErrColumnExists.Code()), "column a already exists"))
```
Comment (Ehco1996) [inline syncer/syncer_test.go:1867]: maybe sqls[0],sqls[1]

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: syncer/syncer_test.go (hunk 2 above)
Comment (lance6716) [reply syncer/syncer_test.go]: https://github.com/pingcap/dm/pull/1949/commits/d1177681f73ded97c75afc27957853dcb717157f

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**